# Introduction

#### My report is focus on a travel guide for people going to visit Toronto but not familiar with there. My work is to simplify various venues and only list the most important and attractive venues. And this simplified neighborhood data will be a brief travel guide for new visitor in Toronto.

# Business Problem

#### Nowadays, people because of the more and more busy work, have restricted time to research a place and make a travel plan before vacation. In addition, as a new visitor it is very difficult to find out best places to have fun or best restaurant to have dinner. Besides, in order to travel on schedule, the transportation and hotel are also very important.
#### Previously, I retrieved all venues in East, West and Central Toronto as below:


In [1]:
import pandas as pd
Toronto_Dict_csv = 'Toronto Venues.csv'   
toronto_venues = pd.read_csv(Toronto_Dict_csv)    # read Toronto Postcode and geospatial data
print(toronto_venues.shape)
toronto_venues.head(10)  # check postcode data

(468, 7)


Neighbourhood  Neighbourhood Latitude  \
0                   The Beaches               43.676357   
1                   The Beaches               43.676357   
2                   The Beaches               43.676357   
3  The Danforth West, Riverdale               43.679557   
4  The Danforth West, Riverdale               43.679557   
5  The Danforth West, Riverdale               43.679557   
6  The Danforth West, Riverdale               43.679557   
7  The Danforth West, Riverdale               43.679557   
8  The Danforth West, Riverdale               43.679557   
9  The Danforth West, Riverdale               43.679557   

   Neighbourhood Longitude                    Venue  Venue Latitude  \
0               -79.293031        Glen Manor Ravine       43.676821   
1               -79.293031        Glen Stewart Park       43.675278   
2               -79.293031      Glen Stewart Ravine       43.676300   
3               -79.352188                 Pantheon       43.677621   
4               -79.352188             Dolce Gelato       43.677773   
5               -79.352188            MenEssentials       43.677820   
6               -79.352188               La Diperie       43.677530   
7               -79.352188  Messini Authentic Gyros       43.677827   
8               -79.352188     Moksha Yoga Danforth       43.677622   
9               -79.352188   Louis Cifer Brew Works       43.677663   

   Venue Longitude        Venue Category  
0       -79.293942                 Trail  
1       -79.294647                  Park  
2       -79.294784  Other Great Outdoors  
3       -79.351434      Greek Restaurant  
4       -79.351187        Ice Cream Shop  
5       -79.351265        Cosmetics Shop  
6       -79.352295        Ice Cream Shop  
7       -79.350569      Greek Restaurant  
8       -79.352116           Yoga Studio  
9       -79.351313               Brewery

In [2]:
print('There are {} uniques categories.'.format(len(toronto_venues['Venue Category'].unique())))

There are 150 uniques categories.


#### We can see there are 150 categories contain Trail, Park, Other Great Outdoors and Ice Cream Shop, et al. So many different venue categories confuse people and make it hard to make a travel plan. Thus, simplifying the venue categories is necessary for new visitors.
#### The first step is ignoring the unnecessary venue, and then classify the rest categories into 6 groups: Entertainment, Food, Hotel, Scenic spot, Shop, Transportation. More details are described in Data Section.

# Data 

#### Pick up the useful data and classify them

In [3]:
Transportation=['Airport Lounge','Bus Stop','Light Rail Station','Lounge','Trail']
Scenic_spot=['Art Gallery','Church','Concert Hall','Construction & Landscaping','Fountain','Garden','History Museum',
            'Lake','Museum','Outdoor Sculpture','Park','Performing Arts Venue','Plaza','Scenic Lookout']
Food=['American Restaurant','Asian Restaurant','BBQ Joint','Bakery','Belgian Restaurant',
      'Breakfast Spot','Bubble Tea Shop','Burger Joint','Burrito Place','Café','Candy Store','Caribbean Restaurant',
     'Coffee Shop','Colombian Restaurant','Comfort Food Restaurant','Cuban Restaurant','Dessert Shop','Diner',
     'Ethiopian Restaurant','Falafel Restaurant','Fast Food Restaurant','Food','Food & Drink Shop','Food Court','Food Truck',
     'French Restaurant','Gluten-free Restaurant','Greek Restaurant','Ice Cream Shop','Indian Restaurant','Italian Restaurant',
     'Japanese Restaurant','Korean Restaurant','Latin American Restaurant','Malay Restaurant','Mexican Restaurant',
     'Middle Eastern Restaurant','New American Restaurant','Noodle House','Pizza Place','Ramen Restaurant','Restaurant',
     'Seafood Restaurant','Steakhouse','Sushi Restaurant','Taco Place','Thai Restaurant','Theme Restaurant',
     'Vegetarian / Vegan Restaurant']
Hotel=['Hostel','Hotel','Pub']
Shop=['Adult Boutique','Clothing Store','Costume Shop','Fish & Chips Shop','Grocery Store','Supermarket',
     'Toy / Game Store']
Entertainment=['Bar','Beer Bar','Cocktail Bar','Dance Studio','Dive Bar','Gastropub','Gay Bar','General Entertainment',
              'Gym','Hotel Bar','Jazz Club','Movie Theater','Sports Bar','Theater','Wine Bar']

In [48]:
toronto_venues_temp=toronto_venues
tot_class=Transportation+Scenic_spot+Food+Hotel+Shop+Entertainment
toronto_venues_temp=toronto_venues_temp[toronto_venues_temp['Venue Category'].isin(tot_class)]
print(toronto_venues_temp.shape)
toronto_venues_temp.head()

(357, 7)


Neighbourhood  Neighbourhood Latitude  \
0                   The Beaches               43.676357   
1                   The Beaches               43.676357   
3  The Danforth West, Riverdale               43.679557   
4  The Danforth West, Riverdale               43.679557   
6  The Danforth West, Riverdale               43.679557   

   Neighbourhood Longitude              Venue  Venue Latitude  \
0               -79.293031  Glen Manor Ravine       43.676821   
1               -79.293031  Glen Stewart Park       43.675278   
3               -79.352188           Pantheon       43.677621   
4               -79.352188       Dolce Gelato       43.677773   
6               -79.352188         La Diperie       43.677530   

   Venue Longitude    Venue Category  
0       -79.293942             Trail  
1       -79.294647              Park  
3       -79.351434  Greek Restaurant  
4       -79.351187    Ice Cream Shop  
6       -79.352295    Ice Cream Shop

#### The unique venue categories reduced from 150 to 93

In [9]:
print('There are {} uniques categories.'.format(len(toronto_venues_temp['Venue Category'].unique())))

There are 93 uniques categories.


#### Classify the new venue categories

In [44]:
toronto_category=toronto_venues_temp[:]

In [52]:

toronto_category['Venue Category'].replace(Food,'Food')
toronto_category['Venue Category'].replace(Transportation,'Transportation')
toronto_category['Venue Category'].replace(Scenic_spot,'Scenic_spot')
toronto_category['Venue Category'].replace(Entertainment,'Entertainment')
toronto_category['Venue Category'].replace(Shop,'Shop')
toronto_category['Venue Category'].replace(Hotel,'Hotel')
print(toronto_category.shape)
toronto_category.head()

(357, 7)


Neighbourhood  Neighbourhood Latitude  \
0                   The Beaches               43.676357   
1                   The Beaches               43.676357   
3  The Danforth West, Riverdale               43.679557   
4  The Danforth West, Riverdale               43.679557   
6  The Danforth West, Riverdale               43.679557   

   Neighbourhood Longitude              Venue  Venue Latitude  \
0               -79.293031  Glen Manor Ravine       43.676821   
1               -79.293031  Glen Stewart Park       43.675278   
3               -79.352188           Pantheon       43.677621   
4               -79.352188       Dolce Gelato       43.677773   
6               -79.352188         La Diperie       43.677530   

   Venue Longitude  Venue Category  
0       -79.293942  Transportation  
1       -79.294647     Scenic_spot  
3       -79.351434            Food  
4       -79.351187            Food  
6       -79.352295            Food

In [60]:
toronto_venues_temp.head()

Neighbourhood  Neighbourhood Latitude  \
0                   The Beaches               43.676357   
1                   The Beaches               43.676357   
3  The Danforth West, Riverdale               43.679557   
4  The Danforth West, Riverdale               43.679557   
6  The Danforth West, Riverdale               43.679557   

   Neighbourhood Longitude              Venue  Venue Latitude  \
0               -79.293031  Glen Manor Ravine       43.676821   
1               -79.293031  Glen Stewart Park       43.675278   
3               -79.352188           Pantheon       43.677621   
4               -79.352188       Dolce Gelato       43.677773   
6               -79.352188         La Diperie       43.677530   

   Venue Longitude    Venue Category  
0       -79.293942             Trail  
1       -79.294647              Park  
3       -79.351434  Greek Restaurant  
4       -79.351187    Ice Cream Shop  
6       -79.352295    Ice Cream Shop

In [11]:
# one hot encoding
toronto_category_onehot = pd.get_dummies(toronto_category[['Venue Category']], prefix="", prefix_sep="")

# add neighborhood column back to dataframe
toronto_category_onehot['Neighbourhood'] = toronto_category['Neighbourhood'] 

# move neighborhood column to the first column
fixed_columns = [toronto_category_onehot.columns[-1]] + list(toronto_category_onehot.columns[:-1])
toronto_category_onehot = toronto_category_onehot[fixed_columns]
print(toronto_category_onehot.shape)
toronto_category_onehot.head()

(357, 7)


Neighbourhood  Entertainment  Food  Hotel  Scenic_spot  \
0                   The Beaches              0     0      0            0   
1                   The Beaches              0     0      0            1   
3  The Danforth West, Riverdale              0     1      0            0   
4  The Danforth West, Riverdale              0     1      0            0   
6  The Danforth West, Riverdale              0     1      0            0   

   Shop  Transportation  
0     0               1  
1     0               0  
3     0               0  
4     0               0  
6     0               0

#### Now, the neighbourhood data is very clear and succinct for new visitor

# Methodology

#### Group rows by neighbourhood and by taking the mean of the frequency of occurrence of each category

In [20]:
import numpy as np # library to handle data in a vectorized manner
import pandas as pd # library for data analsysis

pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

import json # library to handle JSON files


from geopy.geocoders import Nominatim  #convert an address into latitude and longitude values

import requests # library to handle requests
from pandas.io.json import json_normalize # tranform JSON file into a pandas dataframe

# Matplotlib and associated plotting modules
import matplotlib.cm as cm
import matplotlib.colors as colors

# import k-means from clustering stage
from sklearn.cluster import KMeans


import folium # map rendering library

print('Libraries imported.')

Libraries imported.


In [15]:
toronto_grouped = toronto_category_onehot.groupby('Neighbourhood').mean().reset_index()
print(toronto_grouped.shape)
toronto_grouped.head()        # check grouped data

(34, 7)


Neighbourhood  Entertainment      Food  \
0                           Adelaide, King, Richmond       0.090909  0.681818   
1                                        Berczy Park       0.000000  1.000000   
2       Brockton, Exhibition Place, Parkdale Village       0.000000  0.750000   
3  Business Reply Mail Processing Centre 969 Eastern       0.000000  0.000000   
4  CN Tower, Bathurst Quay, Island airport, Harbo...       0.000000  0.000000   

      Hotel  Scenic_spot  Shop  Transportation  
0  0.090909     0.090909   0.0        0.045455  
1  0.000000     0.000000   0.0        0.000000  
2  0.000000     0.125000   0.0        0.125000  
3  0.000000     1.000000   0.0        0.000000  
4  0.000000     0.500000   0.0        0.500000

#### Print each neighbourhood along with all venues

In [17]:
num_top_venues = 6    # Now, there are total 6 venues
for neigh in toronto_grouped['Neighbourhood']:
    print("----"+neigh+"----")
    temp = toronto_grouped[toronto_grouped['Neighbourhood'] == neigh].T.reset_index()
    temp.columns = ['venue','freq']
    temp = temp.iloc[1:]
    temp['freq'] = temp['freq'].astype(float)
    temp = temp.round({'freq': 2})
    print(temp.sort_values('freq', ascending=False).reset_index(drop=True).head(num_top_venues))
    print('\n')

----Adelaide, King, Richmond----
            venue  freq
0            Food  0.68
1   Entertainment  0.09
2           Hotel  0.09
3     Scenic_spot  0.09
4  Transportation  0.05
5            Shop  0.00


----Berczy Park----
            venue  freq
0            Food   1.0
1   Entertainment   0.0
2           Hotel   0.0
3     Scenic_spot   0.0
4            Shop   0.0
5  Transportation   0.0


----Brockton, Exhibition Place, Parkdale Village----
            venue  freq
0            Food  0.75
1     Scenic_spot  0.12
2  Transportation  0.12
3   Entertainment  0.00
4           Hotel  0.00
5            Shop  0.00


----Business Reply Mail Processing Centre 969 Eastern----
            venue  freq
0     Scenic_spot   1.0
1   Entertainment   0.0
2            Food   0.0
3           Hotel   0.0
4            Shop   0.0
5  Transportation   0.0


----CN Tower, Bathurst Quay, Island airport, Harbourfront West, King and Spadina, Railway Lands, South Niagara----
            venue  freq
0     Scenic_spot

#### Sort the venues in descending order

In [18]:
# function to sort the venues in descending order
def return_most_common_venues(row, num_top_venues):
    row_categories = row.iloc[1:]
    row_categories_sorted = row_categories.sort_values(ascending=False)
    
    return row_categories_sorted.index.values[0:num_top_venues]

In [21]:
num_top_venues = 6

indicators = ['st', 'nd', 'rd']

# create columns according to number of top venues
columns = ['Neighbourhood']
for ind in np.arange(num_top_venues):
    try:
        columns.append('{}{} Most Common Venue'.format(ind+1, indicators[ind]))
    except:
        columns.append('{}th Most Common Venue'.format(ind+1))

# create a new dataframe
neighborhoods_venues_sorted = pd.DataFrame(columns=columns)
neighborhoods_venues_sorted['Neighbourhood'] = toronto_grouped['Neighbourhood']

for ind in np.arange(toronto_grouped.shape[0]):
    neighborhoods_venues_sorted.iloc[ind, 1:] = return_most_common_venues(toronto_grouped.iloc[ind, :], num_top_venues)

print(neighborhoods_venues_sorted.shape)
neighborhoods_venues_sorted.head()   # check sorted venues data

(34, 7)


Neighbourhood 1st Most Common Venue  \
0                           Adelaide, King, Richmond                  Food   
1                                        Berczy Park                  Food   
2       Brockton, Exhibition Place, Parkdale Village                  Food   
3  Business Reply Mail Processing Centre 969 Eastern           Scenic_spot   
4  CN Tower, Bathurst Quay, Island airport, Harbo...        Transportation   

  2nd Most Common Venue 3rd Most Common Venue 4th Most Common Venue  \
0           Scenic_spot                 Hotel         Entertainment   
1        Transportation                  Shop           Scenic_spot   
2        Transportation           Scenic_spot                  Shop   
3        Transportation                  Shop                 Hotel   
4           Scenic_spot                  Shop                 Hotel   

  5th Most Common Venue 6th Most Common Venue  
0        Transportation                  Shop  
1                 Hotel         Entertainment  
2                 Hotel         Entertainment  
3                  Food         Entertainment  
4                  Food         Entertainment

#### Cluster Neighbourhoods

In [25]:
# set number of clusters
kclusters = 5

toronto_grouped_clustering = toronto_grouped.drop('Neighbourhood', 1)
# run k-means clustering
kmeans = KMeans(n_clusters=kclusters, random_state=1).fit(toronto_grouped_clustering)

# check cluster labels generated for each row in the dataframe
print(kmeans.labels_[0:10])
print(len(kmeans.labels_))

[0 4 4 2 3 4 4 0 4 0]
34


#### Merge data

In [54]:
Toronto_Dict_csv = 'Toronto Dict.csv'   
toronto_dict = pd.read_csv(Toronto_Dict_csv)    # read Toronto Postcode and geospatial data
print(toronto_dict.shape)
toronto_dict.head()  # check postcode data

(103, 6)


Postcode      Borough                           Neighbourhood Postal Code  \
0      M1B  Scarborough                          Rouge, Malvern         M1B   
1      M1C  Scarborough  Highland Creek, Rouge Hill, Port Union         M1C   
2      M1E  Scarborough       Guildwood, Morningside, West Hill         M1E   
3      M1G  Scarborough                                  Woburn         M1G   
4      M1H  Scarborough                               Cedarbrae         M1H   

    Latitude  Longitude  
0  43.806686 -79.194353  
1  43.784535 -79.160497  
2  43.763573 -79.188711  
3  43.770992 -79.216917  
4  43.773136 -79.239476

In [57]:
toronto_data = toronto_dict[toronto_dict['Borough'].str.contains("Toronto")].reset_index(drop=True)
print(toronto_data.shape)
toronto_data.head()

(38, 6)


Postcode          Borough                   Neighbourhood Postal Code  \
0      M4E     East Toronto                     The Beaches         M4E   
1      M4K     East Toronto    The Danforth West, Riverdale         M4K   
2      M4L     East Toronto  The Beaches West, India Bazaar         M4L   
3      M4M     East Toronto                 Studio District         M4M   
4      M4N  Central Toronto                   Lawrence Park         M4N   

    Latitude  Longitude  
0  43.676357 -79.293031  
1  43.679557 -79.352188  
2  43.668999 -79.315572  
3  43.659526 -79.340923  
4  43.728020 -79.388790

In [71]:
geocsv_data = pd.read_csv('toronto_m.geospatial_data.csv')
print(geocsv_data.shape)
geocsv_data.head()       # check geo data

(103, 3)


Postal Code   Latitude  Longitude
0         M1B  43.806686 -79.194353
1         M1C  43.784535 -79.160497
2         M1E  43.763573 -79.188711
3         M1G  43.770992 -79.216917
4         M1H  43.773136 -79.239476

In [74]:
toronto_di=neighborhoods_venues_sorted.join(geocsv_data)  # merge Toronto sorted venues and geospatial data
print(toronto_di.shape)
toronto_di.head()

(34, 10)


Neighbourhood 1st Most Common Venue  \
0                           Adelaide, King, Richmond                  Food   
1                                        Berczy Park                  Food   
2       Brockton, Exhibition Place, Parkdale Village                  Food   
3  Business Reply Mail Processing Centre 969 Eastern           Scenic_spot   
4  CN Tower, Bathurst Quay, Island airport, Harbo...        Transportation   

  2nd Most Common Venue 3rd Most Common Venue 4th Most Common Venue  \
0           Scenic_spot                 Hotel         Entertainment   
1        Transportation                  Shop           Scenic_spot   
2        Transportation           Scenic_spot                  Shop   
3        Transportation                  Shop                 Hotel   
4           Scenic_spot                  Shop                 Hotel   

  5th Most Common Venue 6th Most Common Venue Postal Code   Latitude  \
0        Transportation                  Shop         M1B  43.806686   
1                 Hotel         Entertainment         M1C  43.784535   
2                 Hotel         Entertainment         M1E  43.763573   
3                  Food         Entertainment         M1G  43.770992   
4                  Food         Entertainment         M1H  43.773136   

   Longitude  
0 -79.194353  
1 -79.160497  
2 -79.188711  
3 -79.216917  
4 -79.239476

In [80]:
toronto_merged = toronto_di
toronto_merged['Cluster Labels'] = kmeans.labels_    # add clustering labels

# merge toronto_grouped with toronto_data to add latitude/longitude for each neighborhood
toronto_merged=toronto_merged.reindex(columns=['Neighbourhood','Postal Code','Latitude','Longitude','Cluster Labels','1st Most Common Venue',
                                '2nd Most Common Venue','3rd Most Common Venue','4th Most Common Venue','5th Most Common Venue',
                               '6th Most Common Venue'])

toronto_merged.head()  

Neighbourhood Postal Code   Latitude  \
0                           Adelaide, King, Richmond         M1B  43.806686   
1                                        Berczy Park         M1C  43.784535   
2       Brockton, Exhibition Place, Parkdale Village         M1E  43.763573   
3  Business Reply Mail Processing Centre 969 Eastern         M1G  43.770992   
4  CN Tower, Bathurst Quay, Island airport, Harbo...         M1H  43.773136   

   Longitude  Cluster Labels 1st Most Common Venue 2nd Most Common Venue  \
0 -79.194353               0                  Food           Scenic_spot   
1 -79.160497               4                  Food        Transportation   
2 -79.188711               4                  Food        Transportation   
3 -79.216917               2           Scenic_spot        Transportation   
4 -79.239476               3        Transportation           Scenic_spot   

  3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
0                 Hotel         Entertainment        Transportation   
1                  Shop           Scenic_spot                 Hotel   
2           Scenic_spot                  Shop                 Hotel   
3                  Shop                 Hotel                  Food   
4                  Shop                 Hotel                  Food   

  6th Most Common Venue  
0                  Shop  
1         Entertainment  
2         Entertainment  
3         Entertainment  
4         Entertainment

In [98]:
map_clusters = folium.Map(location=[43.784535, -79.194353], zoom_start=11)

# set color scheme for the clusters
x = np.arange(kclusters)
ys = [i+x+(i*x)**2 for i in range(kclusters)]
colors_array = cm.rainbow(np.linspace(0, 1, len(ys)))
rainbow = [colors.rgb2hex(i) for i in colors_array]

# add markers to the map
markers_colors = []
for lat, lon, poi, cluster in zip(toronto_merged['Latitude'], toronto_merged['Longitude'], toronto_merged['Neighbourhood'],kmeans.labels_):
    label = folium.Popup(str(poi) + ' Cluster ' + str(cluster), parse_html=True)
    folium.CircleMarker(
        [lat, lon],
        radius=5,
        popup=label,
        color=rainbow[cluster-1],
        fill=True,
        fill_color=rainbow[cluster-1],
        fill_opacity=0.7).add_to(map_clusters)
       
map_clusters

#### Cluster 1

In [87]:
toronto_merged.loc[toronto_merged['Cluster Labels'] == 0, toronto_merged.columns[[0] + list(range(2, toronto_merged.shape[1]))]]

Neighbourhood   Latitude  Longitude  \
0                            Adelaide, King, Richmond  43.806686 -79.194353   
7           Chinatown, Grange Park, Kensington Market  43.711112 -79.284577   
9                                Church and Wellesley  43.692657 -79.264848   
10                     Commerce Court, Victoria Hotel  43.757410 -79.273304   
14           Design Exchange, Toronto Dominion Centre  43.815252 -79.284577   
15                       Dovercourt Village, Dufferin  43.799525 -79.318389   
16             First Canadian Place, Underground city  43.836125 -79.205636   
18  Harbourfront East, Toronto Islands, Union Station  43.778517 -79.346556   
21                           Little Portugal, Trinity  43.789053 -79.408493   
26                           Ryerson, Garden District  43.745906 -79.352188   
27                                     St. James Town  43.725900 -79.340923   
28                    Stn A PO Boxes 25 The Esplanade  43.754328 -79.442259   
29                                    Studio District  43.767980 -79.487262   

    Cluster Labels 1st Most Common Venue 2nd Most Common Venue  \
0                0                  Food           Scenic_spot   
7                0                  Food         Entertainment   
9                0                  Food         Entertainment   
10               0                  Food         Entertainment   
14               0                  Food         Entertainment   
15               0                  Shop                  Food   
16               0                  Food         Entertainment   
18               0                  Food         Entertainment   
21               0                  Food         Entertainment   
26               0                  Food         Entertainment   
27               0                  Food                 Hotel   
28               0                  Food         Entertainment   
29               0                  Food         Entertainment   

   3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
0                  Hotel         Entertainment        Transportation   
7         Transportation                  Shop           Scenic_spot   
9                   Shop           Scenic_spot        Transportation   
10           Scenic_spot                 Hotel        Transportation   
14                 Hotel           Scenic_spot        Transportation   
15         Entertainment        Transportation           Scenic_spot   
16           Scenic_spot        Transportation                  Shop   
18           Scenic_spot                  Shop        Transportation   
21           Scenic_spot        Transportation                  Shop   
26           Scenic_spot                  Shop        Transportation   
27         Entertainment           Scenic_spot        Transportation   
28           Scenic_spot                  Shop                 Hotel   
29                  Shop        Transportation           Scenic_spot   

   6th Most Common Venue  
0                   Shop  
7                  Hotel  
9                  Hotel  
10                  Shop  
14                  Shop  
15                 Hotel  
16                 Hotel  
18                 Hotel  
21                 Hotel  
26                 Hotel  
27                  Shop  
28        Transportation  
29                 Hotel

#### Cluster 2

In [88]:
toronto_merged.loc[toronto_merged['Cluster Labels'] == 1, toronto_merged.columns[[0] + list(range(2, toronto_merged.shape[1]))]]

Neighbourhood   Latitude  Longitude  \
12                                   Davisville North  43.794200 -79.262029   
13  Deer Park, Forest Hill SE, Rathnelly, South Hi...  43.781638 -79.304302   
30                The Annex, North Midtown, Yorkville  43.737473 -79.464763   
32                     The Beaches West, India Bazaar  43.728496 -79.495697   

    Cluster Labels 1st Most Common Venue 2nd Most Common Venue  \
12               1                  Shop           Scenic_spot   
13               1                  Shop                  Food   
30               1                  Shop        Transportation   
32               1                  Shop           Scenic_spot   

   3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
12                  Food        Transportation                 Hotel   
13        Transportation           Scenic_spot                 Hotel   
30           Scenic_spot                 Hotel                  Food   
32        Transportation                 Hotel                  Food   

   6th Most Common Venue  
12         Entertainment  
13         Entertainment  
30         Entertainment  
32         Entertainment

#### Cluster 3

In [91]:
toronto_merged.loc[toronto_merged['Cluster Labels'] == 2, toronto_merged.columns[[0] + list(range(2, toronto_merged.shape[1]))]]

Neighbourhood   Latitude  Longitude  \
3   Business Reply Mail Processing Centre 969 Eastern  43.770992 -79.216917   
20                                      Lawrence Park  43.757490 -79.374714   
22                        Moore Park, Summerhill East  43.770120 -79.408493   

    Cluster Labels 1st Most Common Venue 2nd Most Common Venue  \
3                2           Scenic_spot        Transportation   
20               2           Scenic_spot        Transportation   
22               2           Scenic_spot        Transportation   

   3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
3                   Shop                 Hotel                  Food   
20                  Shop                 Hotel                  Food   
22                  Shop                 Hotel                  Food   

   6th Most Common Venue  
3          Entertainment  
20         Entertainment  
22         Entertainment

#### Cluster 4

In [92]:
toronto_merged.loc[toronto_merged['Cluster Labels'] == 3, toronto_merged.columns[[0] + list(range(2, toronto_merged.shape[1]))]]

Neighbourhood   Latitude  Longitude  \
4   CN Tower, Bathurst Quay, Island airport, Harbo...  43.773136 -79.239476   
24                             Parkdale, Roncesvalles  43.782736 -79.442259   
31                                        The Beaches  43.739015 -79.506944   

    Cluster Labels 1st Most Common Venue 2nd Most Common Venue  \
4                3        Transportation           Scenic_spot   
24               3        Transportation           Scenic_spot   
31               3        Transportation           Scenic_spot   

   3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
4                   Shop                 Hotel                  Food   
24                  Shop                 Hotel                  Food   
31                  Shop                 Hotel                  Food   

   6th Most Common Venue  
4          Entertainment  
24         Entertainment  
31         Entertainment

#### Cluster 5

In [94]:
toronto_merged.loc[toronto_merged['Cluster Labels'] == 4, toronto_merged.columns[[0] + list(range(2, toronto_merged.shape[1]))]]

Neighbourhood   Latitude  Longitude  \
1                                    Berczy Park  43.784535 -79.160497   
2   Brockton, Exhibition Place, Parkdale Village  43.763573 -79.188711   
5                    Cabbagetown, St. James Town  43.744734 -79.239476   
6                             Central Bay Street  43.727929 -79.262029   
8                                       Christie  43.716316 -79.239476   
11                                    Davisville  43.750072 -79.295849   
17                Harbord, University of Toronto  43.803762 -79.363452   
19                     Harbourfront, Regent Park  43.786947 -79.385975   
23                            North Toronto West  43.752758 -79.400049   
25                            Runnymede, Swansea  43.753259 -79.329656   
33                  The Danforth West, Riverdale  43.761631 -79.520999   

    Cluster Labels 1st Most Common Venue 2nd Most Common Venue  \
1                4                  Food        Transportation   
2                4                  Food        Transportation   
5                4                  Food         Entertainment   
6                4                  Food        Transportation   
8                4                  Food                  Shop   
11               4                  Food                  Shop   
17               4                  Food                 Hotel   
19               4                  Food           Scenic_spot   
23               4                  Food        Transportation   
25               4                  Food                  Shop   
33               4                  Food        Transportation   

   3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
1                   Shop           Scenic_spot                 Hotel   
2            Scenic_spot                  Shop                 Hotel   
5            Scenic_spot        Transportation                  Shop   
6                   Shop           Scenic_spot                 Hotel   
8         Transportation           Scenic_spot                 Hotel   
11        Transportation           Scenic_spot                 Hotel   
17        Transportation                  Shop           Scenic_spot   
19        Transportation                  Shop                 Hotel   
23                  Shop           Scenic_spot                 Hotel   
25           Scenic_spot        Transportation                 Hotel   
33                  Shop           Scenic_spot                 Hotel   

   6th Most Common Venue  
1          Entertainment  
2          Entertainment  
5                  Hotel  
6          Entertainment  
8          Entertainment  
11         Entertainment  
17         Entertainment  
19         Entertainment  
23         Entertainment  
25         Entertainment  
33         Entertainment

# Result 

#### I simplify the number of unique venue categories in Toronto from 150 to 93. Then I further calssify the new venue categories into 6 types: Transportation, Scenic spot, Entertainment, Shop and Hotel. This simplified venue categoreis briefly list the most important and attractive venues for new visitor in Toronto. And neighborhoods are clustered and segemented by k means algorithm.

# Discussion

#### From the result, new visitors can make their travel plan and choose which place to stay by their interests. If they prefer to take more time to enjoy the scince spot in Toronto, they can choose to live in the hotel nearby scince spot. If they can't wait to taste the delicious food, they can spend more time in the place arrounded by resturants. Besides, for the people who change the schedule and need to leave next day, they can pick the hotel close to transportation to stay. 

# Conclusion

#### In order to make data clear and brief, we should drop some less important data and further classify them. And we can cluster and segement data by k means algorithm and get the information similar to what we most care or we want.